x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.tsx

```diff
@@ -26,20 +29,82 @@ import {
 } from '../../state_syncing/global_sync';
 import { AppStateManager } from '../../state_syncing/app_state_manager';
 import { startAppStateSyncing } from '../../state_syncing/app_sync';
-import { esFilters } from '../../../../../../../src/plugins/data/public';
+import {
+  esFilters,
+  Filter,
+  Query,
+  TimeRange,
+  IndexPattern,
+  SavedQuery,
+  QueryStateChange,
+  QueryState,
+} from '../../../../../../../src/plugins/data/public';
 import { MapContainer } from '../../../connected_components/map_container';
 import { getIndexPatternsFromIds } from '../../../index_pattern_util';
 import { getTopNavConfig } from './top_nav_config';
 import { getBreadcrumbs, unsavedChangesWarning } from './get_breadcrumbs';
+import {
+  LayerDescriptor,
+  MapRefreshConfig,
+  MapCenterAndZoom,
+  MapQuery,
+} from '../../../../common/descriptor_types';
+import { MapSettings } from '../../../reducers/map';
+
+interface Props {
+  savedMap: any;
+  onAppLeave: AppMountParameters['onAppLeave'];
+  stateTransfer: EmbeddableStateTransfer;
+  originatingApp?: string;
+  layerListConfigOnly: unknown;
+  replaceLayerList: (layerList: LayerDescriptor[]) => any;
+  filters: Filter[];
+  isFullScreen: boolean;
+  isOpenSettingsDisabled: boolean;
+  enableFullScreen: () => any;
+  openMapSettings: () => any;
+  inspectorAdapters: Adapters;
+  nextIndexPatternIds: string[];
+  dispatchSetQuery: ({
+    forceRefresh,
+    filters,
+    query,
+    timeFilters,
+  }: {
+    filters?: Filter[];
+    query?: Query;
+    timeFilters?: TimeRange;
+    forceRefresh?: boolean;
+  }) => void;
+  timeFilters: TimeRange;
+  refreshConfig: MapRefreshConfig;
+  setRefreshConfig: (refreshConfig: MapRefreshConfig) => any;
+  isSaveDisabled: boolean;
+  clearUi: () => void;
+  setGotoWithCenter: (latLonZoom: MapCenterAndZoom) => any;
+  setMapSettings: (mapSettings: MapSettings) => any;
+  setIsLayerTOCOpen: (isLayerTOCOpen: boolean) => any;
+  setOpenTOCDetails: (openTOCDetails: string[]) => any;
+  query: MapQuery | undefined;
+}
 
-export class MapsAppView extends React.Component {
-  _globalSyncUnsubscribe = null;
-  _globalSyncChangeMonitorSubscription = null;
-  _appSyncUnsubscribe = null;
+interface State {
+  initialized: boolean;
+  initialLayerListConfig: any;
+  indexPatterns: IndexPattern[];
+  savedQuery: SavedQuery | string;
```

Comment:
Can savedQuery ever be a string?

Reply:
@nreese The only place `savedQuery` was string is in the [state initialization](https://github.com/elastic/kibana/blob/765f772c4ea358eb9fff43f2ea2558239d507a17/x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.tsx#L112 ). I wasn't sure whether there are any string checks for `savedQuery` that is why added string type too.  I can remove that to be `savedQuery?: SavedQuery` and change the `savedQuery` in state initialization to `undefined`. Is that okay?

Reply:
Changing savedQuery initialization to will clear things up. It is a bug that this was initialized to an empty string
```
{
    query: '',
    language: getUiSettings().get('search:queryLanguage'),
}
```